src/serializer/functions.js

```diff
@@ -151,6 +172,22 @@ export class Functions {
     applyOptimizedReactComponents(this.realm, this.writeEffects, environmentRecordIdAfterGlobalCode);
   }
 
+  // When __optimize calls are called, an optional "pure" property can be passed
+  // on a config object to disable pure scope, otherwise pure scope is enabled by default.
+  // i.e. __optimize(foo, { pure: false });
+  _isPureOptimizedFunction(optimizedFunctionConfig: void | Value): boolean {
```

Comment:
I would add a comment here that in pure functions, we assume that any throw not surrounded by a try block within that pure scope is assumed to be uncaught and exit the entire program (or possibly add a flag for that aside from the pure flag?).